### PR TITLE
use SubElement where possible when creating dom trees

### PR DESCRIPTION
**Pull request by deichmab-draeger**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have updated the [changelog](../CHANGELOG.md) accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: src/sdc11073/mdib/mdibbase.py [hunk 1]
```diff
@@ -358,32 +358,34 @@ def _reconstruct_md_description(self) -> xml_utils.LxmlElement:
...
+                                 attrib={'Handle': descriptor_container.Handle},
+                                 nsmap=ns_map)
         descriptor_container.update_node(node, self.nsmapper, set_xsi_type)  # create all
         child_list = self.descriptions.parent_handle.get(descriptor_container.Handle, [])
         # append all child containers, then bring all child elements in correct order
         for child in child_list:
             child_tag, set_xsi = descriptor_container.tag_name_for_child_descriptor(child.NODETYPE)
-            child_node = self.make_descriptor_node(child, child_tag, set_xsi)
+            child_node = self.make_descriptor_node(child, node, child_tag, set_xsi)
             node.append(child_node)
```
Comment (a-kleinf) [inline src/sdc11073/mdib/mdibbase.py:389]: I think this must be removed

Comment (deichmab-draeger) [reply src/sdc11073/mdib/mdibbase.py]: done

File: tests/test_mdib.py [hunk 2]
```diff
@@ -122,3 +123,86 @@ def test_get_mixed_states(self):
...
+        for test_data in mdibs:
+            # parse mdib data into container and reconstruct mdib data back to a msg:GetMdibResponse
+            # so that it can be validated by xml schema validator
+            mdib_text = test_data.mdib_text.encode('utf-8')
+            mdib_container = ProviderMdib.from_string(mdib_text)
+            mdib_node, mdib_version_group = mdib_container.reconstruct_mdib_with_context_states()
+            arg_nodes = mdib_node.xpath('//*/pm:Arg', namespaces={'pm': "http://standards.ieee.org/downloads/11073/11073-10207-2017/participant"})
+            arg_node = arg_nodes[0]
+            prefix = arg_node.text.split(':')[0]
+            self.assertTrue('msg' in arg_node.nsmap)
```
Comment (a-kleinf) [inline tests/test_mdib.py:207]: is this needed?

File: tests/test_mdib.py [hunk 3]
```diff
@@ -122,3 +123,86 @@ def test_get_mixed_states(self):
...
+            self.assertTrue(prefix in arg_node.nsmap)
```
Comment (a-kleinf) [inline tests/test_mdib.py:208]: maybe add
            self.assertEqual(test_data.expected_qname.namespace, arg_node.nsmap[prefix])
            self.assertEqual(test_data.expected_qname.localname, arg_node.text.split(':')[1])

**Review by a-kleinf — CHANGES_REQUESTED** [review]

File: tests/test_mdib.py (hunk 2 above)
Comment (deichmab-draeger) [reply tests/test_mdib.py:207]: no

File: tests/test_mdib.py (hunk 3 above)
Comment (deichmab-draeger) [reply tests/test_mdib.py:210]: done

**Review by a-kleinf — APPROVED** [review]

**Review by a-kleinf — APPROVED** [review]

**Pull request merged**